datalad/support/gitrepo.py

```diff
@@ -1977,16 +1977,18 @@ def get_merge_base(self, commitishes):
             commitishes = commitishes + [self.get_active_branch()]
 
         try:
-            bases = self.repo.merge_base(*commitishes)
-        except GitCommandError as exc:
-            if "fatal: Not a valid object name" in str(exc):
-                return None
-            raise
+            bases = self.call_git_oneline(['merge-base'] + commitishes)
+        except CommandError as exc:
+            if "fatal: Not a valid object name" not in exc.stderr and (
+                    exc.stdout or exc.stderr):
+                # we don't know what this is, and we have something to tell about it
+                # then log it. Without output, it could also just be "no merge base"
+                lgr.debug(exc_str(exc))
+            return None
```

Comment:
There's a change in behavior here, switching to logging unknown exceptions rather than raising them.  My preference would be to raise the error rather than logging in this situation.  Perhaps something like this:

```diff
diff --git a/datalad/support/gitrepo.py b/datalad/support/gitrepo.py
index 30f08f652..642bfb511 100644
--- a/datalad/support/gitrepo.py
+++ b/datalad/support/gitrepo.py
@@ -1979,12 +1979,12 @@ def get_merge_base(self, commitishes):
         try:
             bases = self.call_git_oneline(['merge-base'] + commitishes)
         except CommandError as exc:
-            if "fatal: Not a valid object name" not in exc.stderr and (
-                    exc.stdout or exc.stderr):
-                # we don't know what this is, and we have something to tell about it
-                # then log it. Without output, it could also just be "no merge base"
-                lgr.debug(exc_str(exc))
-            return None
+            if exc.code == 1 and not (exc.stdout or exc.stderr):
+                # No merge base was found (unrelated commits).
+                return None
+            if "fatal: Not a valid object name" in exc.stderr:
+                return None
+            raise
 
         if not bases:
             return None
```

I don't have a strong objection to leaving it as is, though.

Reply:
I like this more than what I did, will change. Thx!